### PR TITLE
[1LP][RFR] Add Openstack Node lifecycle tests

### PR DESCRIPTION
**Pull request by okolisny**
Adding tests for Openstack nodes' lifecycle operations: scale down, delete, register, introspect, provide, scale out.
Please no PRT run, any wrong thing might wreck your OSP environment. Added test_flag for avoid you of running this on your environments.

### CODE REVIEW

**Comment by okolisny** [comment]
Successful execution in Jenkins https://rhos-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/rhos11_cfme58_integration/48/TEST_REPORT/

File: cfme/common/host_views.py [hunk 1]
```diff
@@ -240,7 +241,7 @@ class HostsView(ComputeInfrastructureHostsView):
 
     @property
     def is_displayed(self):
-        return self.in_compute_infrastructure_hosts and self.title.text == "Hosts"
+        return self.in_compute_infrastructure_hosts and self.title.text in "Hosts / Nodes"
```
Comment (quarckster) [inline cfme/common/host_views.py]: **REQUIRED**
Is it correct `self.title.text in "Hosts / Nodes"`? Maybe `"Hosts / Nodes" in self.title.text`

File: cfme/common/provider_views.py [hunk 2]
```diff
@@ -218,7 +218,35 @@ class ProviderRegisterNodesView(View):
...
+    cancel = Button(value='Cancel')
+
+    @property
+    def is_displayed(self):
+        return False
+
+
+class ProviderScaleDownView(View):
+    """
+     represents Scale down view (exists for Infra OpenStack provider)
```
Comment (quarckster) [inline cfme/common/provider_views.py]: **REQUIRED**
If it exists only in Infra OpenStack provider might be better to place this view into `openstack_infra.py`?

File: cfme/infrastructure/host.py [hunk 3]
```diff
@@ -164,13 +164,8 @@ def delete(self, cancel=True):
...
             cancel (bool): Whether to cancel the deletion, defaults to True
         """
-
         view = navigate_to(self, "Details")
-        view.toolbar.configuration.item_select("Remove item", handle_alert=cancel)
-        if not cancel:
-            view = self.create_view(HostsView)
-            assert view.is_displayed
-            view.flash.assert_success_message("The selected Hosts / Nodes was deleted")
+        view.toolbar.configuration.item_select("Remove item", handle_alert=not cancel)
```
Comment (quarckster) [inline cfme/infrastructure/host.py]: **REQUIRED**
Why did you delete it?

File: cfme/infrastructure/host.py [hunk 4]
```diff
@@ -211,6 +206,11 @@ def power_off(self):
     def get_power_state(self):
         return self.get_detail("Properties", "Power State")
 
+    def provide(self):
+        """Provide Openstack infra node"""
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Provide Node', handle_alert=True)
```
Comment (quarckster) [inline cfme/infrastructure/host.py]: **REQUIRED**
Please do not add specific methods into common class. Inherit it and implement this method there.

Comment (okolisny) [inline cfme/infrastructure/host.py]: @quarckster do you suggest to add class Node(Host) ?

File: cfme/common/host_views.py (hunk 1 above)
Comment (okolisny) [inline cfme/common/host_views.py]: I assume that `self.title.text` can be equal to `"Hosts"` or `"Nodes"` or `"Hosts / Nodes"`. So `self.title.text in "Hosts / Nodes"` handles all the cases.

File: cfme/infrastructure/host.py (hunk 4 above)
Comment (quarckster) [inline cfme/infrastructure/host.py]: @okolisny Yes, maybe even OpenStackNode(Host)

File: cfme/common/provider_views.py (hunk 2 above)
Comment (okolisny) [inline cfme/common/provider_views.py]: Done

File: cfme/infrastructure/host.py (hunk 3 above)
Comment (okolisny) [inline cfme/infrastructure/host.py]: I've returned it back. The reason I had deleted it before is that it hadn't work for Openstack nodes. Fixed it [here](https://github.com/ManageIQ/integration_tests/pull/5152/files#diff-48da398b86a224ed0b471e7540050346R42)

File: cfme/tests/openstack/infrastructure/test_host_lifecycle.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
+import pytest
+
+
```
Comment (quarckster) [inline cfme/tests/openstack/infrastructure/test_host_lifecycle.py]: **REQUIRED**
Extra blank line is here.

File: cfme/tests/openstack/infrastructure/test_host_lifecycle.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+from cfme.exceptions import HostNotFound
+from cfme.infrastructure.host import get_all_hosts
+from cfme.infrastructure.openstack_node import OpenstackNode
+from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from cfme.web_ui import flash, toolbar
```
Comment (quarckster) [inline cfme/tests/openstack/infrastructure/test_host_lifecycle.py]: **REQUIRED**
Old `web_ui` elements shouldn't be use in the new code.

File: cfme/infrastructure/provider/openstack_infra.py [hunk 7]
```diff
@@ -149,10 +169,71 @@ def step(self):
...
+    @property
+    def is_displayed(self):
+        return False
+
+
+class ProviderScaleDownView(View):
+    """
+     represents Scale down view
+    """
+    table = Table(locator='//div[contains(@class, "form-horizontal")]//table')
```
Comment (quarckster) [inline cfme/infrastructure/provider/openstack_infra.py]: **OPTIONAL**
I would rely on some more unique path. Try to find parent div with some id.
